lib/datasource-processor.js

```diff
@@ -459,15 +453,15 @@ function getBands(ds, bandCount){
         nodata = band.noDataValue;
         unitType = ds.bands.get(i).unitType;
         //Error check file sources
-        try { 
+        try {
             bandStats = band.getStatistics(false, true);
         } catch(err){
-            return invalid("Error getting statistics of band. 1 or more of the VRT file's relative sources may be missing: " + err.message);
-        };
+            throw invalid("Error getting statistics of band. 1 or more of the VRT file's relative sources may be missing: " + err.message);
+        }
```

Comment:
Is the `throw` here intentional? It probably is, just double checking since I've noticed most places `return invalid` is used.